### PR TITLE
Merge ostree-ext

### PR DESCRIPTION
**Pull request by cgwalters**
xref https://github.com/ostreedev/ostree-rs-ext/issues/675

Basically I see bootc as the successor/inheritor of the original ostree-container work.

It will help us close the loop faster on things to merge it into this project.

TODO followups:

- [ ] Remove more unreferenced files (e.g. GHA workflows)
- [ ] Add `bootc internals ostree-container` or so we can take over the `ostree container` verb

### CODE REVIEW

**Comment by cgwalters** [comment]
Ah, messed this one up, will redo later

**Comment by cgwalters** [comment]
OK I redid this one, now with proper git history

**Comment by cgwalters** [comment]
Ugh, right I think I need to `git merge --signoff`

File: ostree-ext/.github/workflows/bootc.yml
Comment (p5) [inline ostree-ext/.github/workflows/bootc.yml:1]: Unfortunately, GitHub doesn't support GHA workflows in subdirectories, so these tests will not run.  
If you wish to keep these tests, it's probably best moving the contents of `./ostree-ext/.github/workflows/` to `./.github/workflows/`

Comment (cgwalters) [reply ostree-ext/.github/workflows/bootc.yml:1]: Thanks, yep there's more followup to do here, though in practice we have decent coverage from the unit tests.

**Comment by cgwalters** [comment]
I have clicked "Set DCO to pass" to override the signoff on individual commits. Most of these commits are mine anyways, and with this statement I add my `Signed-off-by` onto all of these commits.

**Review by jeckersb — APPROVED** [review]

**Comment by jeckersb** [comment]
Yay for one less repo to watch :partying_face:

**Pull request merged**